pkg/controller/identity/iamrole/controller.go

```diff
@@ -107,38 +109,39 @@ func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.E
 	}
 
 	if err != nil {
```

Comment:
Nitpick: this error check and the one above can be refactored as:

```
if err != nil {
	return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(iam.IsErrorNotFound, err), errGet)
}
```
`ResourceExists: false` is implied and an error will cause requeue :heavy_check_mark: